### PR TITLE
Backfill only up to MIN_EPOCHS_FOR_BLOCK_REQUESTS blocks

### PR DESCRIPTION
**Pull request by arnetheduck**
When backfilling, we only need to download blocks that are newer than MIN_EPOCHS_FOR_BLOCK_REQUESTS - the rest cannot reliably be fetched from the network and does not have to be provided to others.

This change affects only trusted-node-synced clients - genesis sync continues to work as before (because it needs to construct a state by building it from genesis).

Those wishing to complete a backfill should do so with era files instead.

### CODE REVIEW

**Comment by arnetheduck** [comment]
mac test failure needs investigation, could be something real

**Comment by arnetheduck** [comment]
was something real: pre-merge block production broken without EL

File: beacon_chain/sync/sync_queue.nim [hunk 1]
```diff
@@ -793,7 +783,7 @@ proc push*[T](sq: SyncQueue[T], sr: SyncRequest[T],
           if safeSlot > failSlot:
             let rewindSlot = sq.getRewindPoint(failSlot, safeSlot)
             # It's quite common peers give us fewer blocks than we ask for
-            info "Gap in block range response, rewinding", request = req,
+            debug "Gap in block range response, rewinding", request = req,
```
Comment (tersec) [inline beacon_chain/sync/sync_queue.nim:786]: Is this a general/unrelated change or does this become a more frequent noise-level message after this PR?

File: beacon_chain/nimbus_beacon_node.nim [hunk 2]
```diff
@@ -445,10 +442,7 @@ proc init*(T: type BeaconNode,
   let optJwtSecret = rng[].loadJwtSecret(config, allowCreate = false)
 
   if config.web3Urls.len() == 0:
-    if cfg.BELLATRIX_FORK_EPOCH == FAR_FUTURE_EPOCH:
-      notice "Running without execution client - validator features partially disabled (see https://nimbus.guide/eth1.html)"
-    else:
-      notice "Running without execution client - validator features disabled (see https://nimbus.guide/eth1.html)"
+    notice "Running without execution client - validator features disabled (see https://nimbus.guide/eth1.html)"
```
Comment (tersec) [inline beacon_chain/nimbus_beacon_node.nim:445]: So long as this remains only informational either way, it still allows compatibility with a testnet/devnet which decides not to go into Bellatrix

File: beacon_chain/sync/sync_queue.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/sync/sync_queue.nim:786]: Unrelated, but it's spammy when syncing which I've been doing a lot while testing

File: beacon_chain/nimbus_beacon_node.nim (hunk 2 above)
Comment (arnetheduck) [reply beacon_chain/nimbus_beacon_node.nim:445]: Yeah, there's nuance here about fork timing etc that could be expressed better but .. the old check was also slightly wrong (for the pre-merge period) - not sure it's worth the lines of code to fix properly
